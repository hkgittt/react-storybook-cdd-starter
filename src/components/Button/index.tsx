import React from "react";

interface ButtonProps {
  label?: string;
}

const Button: React.FC<ButtonProps> = ({ label = "Click Me" }) => (
  <button>{label}</button>
);

export default Button;
