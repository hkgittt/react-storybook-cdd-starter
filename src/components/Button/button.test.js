import React from 'react';
import Button from './';
import renderer from 'react-test-renderer';

describe('Sample test', () => {
  it('confirms the obvious', () => {
    expect(true).toEqual(true);
  });

  it('renders google button correctly', () => {
    const tree = renderer
      .create(<Button label="Google" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders facebook button correctly', () => {
    const tree = renderer
      .create(<Button label="Facebook" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
