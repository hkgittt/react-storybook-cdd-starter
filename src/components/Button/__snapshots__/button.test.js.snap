// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sample test renders facebook button correctly 1`] = `
<button>
  Facebook
</button>
`;

exports[`Sample test renders google button correctly 1`] = `
<button>
  Google
</button>
`;
